### PR TITLE
new: Clone | Reset | Delete element feature

### PR DESCRIPTION
**Pull request by rohitp-bsf**
### Which all elements support clone feature.

- Button
- HTML
- Menu
- Divider
- Social
- Widgets

### What does reset do?
- Reset will update the component values to the default like a fresh component drag-dropped.

### What does Delete do?
- Delete will eradicate the element from the customizer which will improve the performance automatically. An eradicated element can not be added further until you clone it from somewhere else component.

### What does cross button click do?
- Cross button will just remove the component from the grid and ultimately from the view but can be added with saved values by clicking on the `+` icon.

### Checklist:
- [x] My code is tested
- [x] My code passes the PHPCS tests
- [x] My code follows accessibility standards 
- [x] My code has proper inline documentation 
- [x] I've included any necessary tests 
- [ ] I've included developer documentation 
- [ ] I've added proper labels to this pull request

### CODE REVIEW

File: inc/assets/js/customizer-preview.js [hunk 1]
```diff
@@ -67,6 +67,11 @@ function astra_builder_advanced_css( section, selector ) {
     // Padding.
     wp.customize( 'astra-settings[' + section + '-padding]', function( value ) {
         value.bind( function( padding ) {
+
+			if( ! padding.hasOwnProperty('desktop') ) {
```
Comment (vrundakansara) [inline inc/assets/js/customizer-preview.js:71]: Should we be checking this for tablet and mobile too?

File: inc/assets/js/customizer-preview.js [hunk 2]
```diff
@@ -107,6 +112,11 @@ function astra_builder_advanced_css( section, selector ) {
     // Margin.
     wp.customize( 'astra-settings[' + section + '-margin]', function( value ) {
         value.bind( function( margin ) {
+
+        	if( ! margin.hasOwnProperty('desktop') ) {
```
Comment (vrundakansara) [inline inc/assets/js/customizer-preview.js:116]: Should we be checking this for tablet and mobile too?

File: inc/builder/markup/class-astra-builder-footer.php [hunk 3]
```diff
@@ -62,21 +62,23 @@ public function __construct() {
 			// Core Components.
 			add_action( 'astra_footer_copyright', array( $this, 'footer_copyright' ), 10 );
 
-			for ( $index = 1; $index <= Astra_Builder_Helper::$num_of_footer_html; $index++ ) {
-				add_action( 'astra_footer_html_' . $index, array( $this, 'footer_html_' . $index ) );
-				self::$methods[] = 'footer_html_' . $index;
-			}
+			for ( $index = 1; $index <= Astra_Builder_Helper::$component_limit; $index++ ) {
```
Comment (vrundakansara) [inline inc/builder/markup/class-astra-builder-footer.php:65]: Do we need to check Astra Pro constant here like it is done here? - https://github.com/brainstormforce/astra/pull/2257/files#diff-539be79abcb46bec35e43928b3f7e8a4794812c524141df1532736d1d08c87d8R80

File: inc/builder/type/base/assets/js/customizer-preview.js [hunk 4]
```diff
@@ -179,6 +184,34 @@ function astra_builder_button_css( builder_type = 'header', button_count ) {
 					astra_add_dynamic_css( 'astra-settings[' + builder_type + '-button'+ index +'-border-size]', dynamicStyle );
 				} );
 			} );
+
+			if( 'footer' == builder_type ) {
```
Comment (vrundakansara) [inline inc/builder/type/base/assets/js/customizer-preview.js:188]: We already have this added here - `wp-content/themes/astra/inc/builder/type/footer/button/assets/js/customizer-preview.js`

File: inc/assets/js/customizer-preview.js (hunk 1 above)
Comment (rohitp-bsf) [reply inc/assets/js/customizer-preview.js:71]: It was the problem of default values. So just added an initial check. There is no possibility of missing mobile and other properties.

I can add it but no need for this here as of now. Please let me know. @vrundakansara

File: inc/assets/js/customizer-preview.js (hunk 2 above)
Comment (rohitp-bsf) [reply inc/assets/js/customizer-preview.js:116]: It was the problem of default values. So just added an initial check. There is no possibility of missing mobile and other properties.

I can add it but no need for this here as of now. Please let me know. @vrundakansara

File: inc/builder/markup/class-astra-builder-footer.php (hunk 3 above)
Comment (rohitp-bsf) [reply inc/builder/markup/class-astra-builder-footer.php:65]: No, we do not need it now. As we are calculating $component count initially on the basis of that we are handling the no of components from the JS itself.

File: inc/builder/type/base/assets/js/customizer-preview.js (hunk 4 above)
Comment (rohitp-bsf) [reply inc/builder/type/base/assets/js/customizer-preview.js:188]: Removed!

**Review by vrundakansara — APPROVED** [review]

**Review by RajkiranBagal — APPROVED** [review]

**Review by vrundakansara — APPROVED** [review]

**Review by vrundakansara — APPROVED** [review]
Ganpati Bappa Morya! 🎉

**Pull request merged**